pkg/metrics/metrics.go

```diff
@@ -96,19 +97,32 @@ func WithInstallIDFileOverride(path string) Option {
 	}
 }
 
+// WithAsyncErrorHandler registers a function to handle any errors occurring
+// during async metric writing. It may be run after the context has
+// timed out, so long-lived blocking operations are discouraged.
```

Comment:
Maybe mention in this comment that this function can run concurrently in multiple goroutines so it should only do things that are thread-safe.

Reply:
Updated